include/librealsense2/dds/dds-device-broadcaster.cpp

```diff
@@ -120,74 +117,71 @@ bool dds_device_broadcaster::run()
     return true;
 }
 
-std::string dds_device_broadcaster::add_device( rs2::device dev )
+void dds_device_broadcaster::add_device( device_info const & dev_info )
 {
-    auto device_serial = dev.get_info( RS2_CAMERA_INFO_SERIAL_NUMBER );
-    std::vector< std::pair< std::string, rs2::device > > devices_to_add;
-    devices_to_add.push_back( { device_serial, dev } );
-
     // Post the connected device
-    handle_device_changes( {}, devices_to_add );
-
-    return get_topic_root( dev.get_info( RS2_CAMERA_INFO_NAME ), dev.get_info( RS2_CAMERA_INFO_SERIAL_NUMBER ) );
+    handle_device_changes( {}, { dev_info } );
 }
 
-void dds_device_broadcaster::remove_device( rs2::device dev )
+void dds_device_broadcaster::remove_device( device_info const & dev_info )
 {
-    auto device_serial = dev.get_info( RS2_CAMERA_INFO_SERIAL_NUMBER );
-    std::vector< std::string > devices_to_remove;
-    devices_to_remove.push_back( device_serial );
-
     // Post the disconnected device
-    handle_device_changes( devices_to_remove, {} );
+    handle_device_changes( { dev_info.serial }, {} );
 }
 
 void dds_device_broadcaster::handle_device_changes(
```

Comment:
handle_device_changes is redundant.
We always call it with one empty list and another with only one device. add_device and remove_device should just implement their logic separately.

Reply:
Yeah but I think the idea was to put the `invoke` and exception handling for both in a common place.
@Nir-Az the `invoke` does need explaining... I think the idea was:
```C++
// We want to be as quick as possible and not wait for DDS in any way -- so we handle
// device changes in the background, using invoke:
```
??

Reply:
You can call invoke from both functions if needed, don't see the added value here it is not long or complicated code. This way is less CLEAN code and also less efficient.

Reply:
I don't think it's less efficient, or less clean. I would have liked it to be straightforward, but the invocation and catching of either is not and is common to both so it makes sense to place in a common function.

Reply:
It did both add / remove at first, but with the implementation it was changed and I only needed it separately .
I left it common since it's not causing any harm and it was already written and tested.
We can separate in future PR's if we like, let's leave it for now :)